src/main/java/com/scottlogic/FullNamePostSorter.java

```diff
@@ -0,0 +1,92 @@
+package com.scottlogic;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.List;
+
+public class FullNamePostSorter implements PostSorter {
+
+    @Override
+    public List<UserPost> sort(List<UserPost> inputList, SortOrder orderIn) {
+
+        List<UserPost> namesNull = new ArrayList<>();
+        List<UserPost> namesSpace = new ArrayList<>();
+        List<UserPost> remainingPosts = new ArrayList<>();
+
+        for (UserPost userPost : inputList) {
+            if (userPost.getAuthor() == null)
+                namesNull.add(userPost);
+            else if (userPost.getAuthor().isBlank())
+                namesSpace.add(userPost);
+            else
+                remainingPosts.add(userPost);
+        }
+        switch (orderIn) {
+
+            case ASC -> Collections.sort(remainingPosts, Comparator.comparing(UserPost::getAuthor, (name1, name2) -> {
+                String firstName1;
+                String firstName2;
+                String lastName1;
+                String lastName2;
+
+                String[] split1 = name1.split(" ");
+                firstName1 = split1[0];
+                if (split1.length == 1) lastName1 = " ";
```

Comment:
I'm not keen of inline if statements like this. If you want to condense your control flow, have a look into the [ternary operator](https://www.baeldung.com/java-ternary-operator).